### ISSUE TITLE
OPX-config-fanout does not support in z9264 platform

### ISSUE THREAD

**Issue by waliulislam** [open-switch/opx-platform-config#58]
Z9264F-ON has 64 100g ports, out of which 32 ports can be fanned out to 10Gx4 or 25Gx4 modes. Since there is a limitation to 128 physical ports, the 64 ports are grouped into 32 "hybrid groups." Hybrid groups need to be set to "restricted" mode in order to support 10Gx4 / 25Gx4 breakout modes, and in restricted mode one of the two ports of that hybrid group will be disabled. Here is an example of how to do 10gx4 breakout of port e101-015-0:
```
root@OPX:~# opx-config-hybrid-group get hybrid-group --name hybrid-group-8
Namespace(command='get', name='hybrid-group-8', subcommand='hybrid-group')
Hybrid Group Name: hybrid-group-8 Profile: unrestricted
Port ID: 16 Phy-mode: ether Breakout-mode: 100gx1
Port ID: 15 Phy-mode: ether Breakout-mode: 100gx1
root@OPX:~# nas_intf_set.py -o delete -t ether e101-015-0
delete
e101-015-0
ether
None
None
Success
Key: 1.19.1245192.
cps/object-group/return-code = 0
dell-base-if-cmn/set-interface/input/operation = 2
if/interfaces/interface/name = e101-015-0
root@OPX:~# nas_intf_set.py -o delete -t ether e101-016-0
delete
e101-016-0
ether
None
None
Success
Key: 1.19.1245192.
root@OPX:~# 
root@OPX:~# opx-config-hybrid-group set hybrid-group profile --name hybrid-group-8 --profile restricted
Namespace(command='set', config='profile', name='hybrid-group-8', profile='restricted', subcommand='hybrid-group')
root@OPX:~# opx-config-hybrid-group get hybrid-group --name hybrid-group-8
Namespace(command='get', name='hybrid-group-8', subcommand='hybrid-group')
Hybrid Group Name: hybrid-group-8 Profile: restricted
Port ID: 16 Phy-mode: ether Breakout-mode: disabled
Port ID: 15 Phy-mode: ether Breakout-mode: 100gx1
root@OPX:~#
root@OPX:~# opx-config-hybrid-group set hybrid-group port --name hybrid-group-8 --port-id 15 --breakout 10gx4
Namespace(breakout='10gx4', command='set', config='port', name='hybrid-group-8', phy_mode='ether', port_id='15', subcommand='hybrid-group')
root@OPX:~#
root@OPX:~#
root@OPX:~# opx-config-hybrid-group get hybrid-group --name hybrid-group-8
Namespace(command='get', name='hybrid-group-8', subcommand='hybrid-group')
Hybrid Group Name: hybrid-group-8 Profile: restricted
Port ID: 16 Phy-mode: ether Breakout-mode: disabled
Port ID: 15 Phy-mode: ether Breakout-mode: 10gx4
root@OPX:~# nas_intf_set.py -o create -t ether e101-015-1
create
e101-015-1
ether
None
None
Success
Key: 1.19.1245192.
base-if-phy/if/interfaces/interface/npu-id = 0
dell-if/if/interfaces/interface/duplex = 1
dell-if/if/interfaces/interface/phys-address = 50:9a:4c:d5:ad:2a
if/interfaces/interface/type = ianaift:ethernetCsmacd
if/interfaces/interface/name = e101-015-1
dell-base-if-cmn/if/interfaces/interface/if-index = 73
dell-if/if/interfaces/interface/auto-negotiation = 1
dell-base-if-cmn/set-interface/input/operation = 1
base-if-phy/hardware-port/front-panel-port = 15
base-if-phy/if/interfaces/interface/hw-profile = 3
cps/object-group/return-code = 0
base-if-phy/hardware-port/subport-id = 1
dell-if/if/interfaces/interface/speed = 4
base-if-phy/if/interfaces/interface/port-id = 113
base-if-phy/if/interfaces/interface/phy-media = 43
base-if-phy/if/interfaces/interface/supported-autoneg = 4
dell-if/if/interfaces/interface/negotiation = 1
root@OPX:~# nas_intf_set.py -o create -t ether e101-015-2
create
e101-015-2
ether
None
None
Success
Key: 1.19.1245192.
base-if-phy/if/interfaces/interface/npu-id = 0
dell-if/if/interfaces/interface/duplex = 1
dell-if/if/interfaces/interface/phys-address = 50:9a:4c:d5:ad:2b
if/interfaces/interface/type = ianaift:ethernetCsmacd
if/interfaces/interface/name = e101-015-2
dell-base-if-cmn/if/interfaces/interface/if-index = 74
dell-if/if/interfaces/interface/auto-negotiation = 1
dell-base-if-cmn/set-interface/input/operation = 1
base-if-phy/hardware-port/front-panel-port = 15
base-if-phy/if/interfaces/interface/hw-profile = 3
cps/object-group/return-code = 0
base-if-phy/hardware-port/subport-id = 2
dell-if/if/interfaces/interface/speed = 4
base-if-phy/if/interfaces/interface/port-id = 114
base-if-phy/if/interfaces/interface/phy-media = 43
base-if-phy/if/interfaces/interface/supported-autoneg = 4
dell-if/if/interfaces/interface/negotiation = 1
root@OPX:~# nas_intf_set.py -o create -t ether e101-015-3
create
e101-015-3
ether
None
None
Success
Key: 1.19.1245192.
base-if-phy/if/interfaces/interface/npu-id = 0
dell-if/if/interfaces/interface/duplex = 1
dell-if/if/interfaces/interface/phys-address = 50:9a:4c:d5:ad:2c
if/interfaces/interface/type = ianaift:ethernetCsmacd
if/interfaces/interface/name = e101-015-3
dell-base-if-cmn/if/interfaces/interface/if-index = 75
dell-if/if/interfaces/interface/auto-negotiation = 1
dell-base-if-cmn/set-interface/input/operation = 1
base-if-phy/hardware-port/front-panel-port = 15
base-if-phy/if/interfaces/interface/hw-profile = 3
cps/object-group/return-code = 0
base-if-phy/hardware-port/subport-id = 3
dell-if/if/interfaces/interface/speed = 4
base-if-phy/if/interfaces/interface/port-id = 115
base-if-phy/if/interfaces/interface/phy-media = 43
base-if-phy/if/interfaces/interface/supported-autoneg = 4
dell-if/if/interfaces/interface/negotiation = 1
root@OPX:~# nas_intf_set.py -o create -t ether e101-015-4
create
e101-015-4
ether
None
None
Success
Key: 1.19.1245192.
base-if-phy/if/interfaces/interface/npu-id = 0
dell-if/if/interfaces/interface/duplex = 1
dell-if/if/interfaces/interface/phys-address = 50:9a:4c:d5:ad:2d
if/interfaces/interface/type = ianaift:ethernetCsmacd
if/interfaces/interface/name = e101-015-4
dell-base-if-cmn/if/interfaces/interface/if-index = 76
dell-if/if/interfaces/interface/auto-negotiation = 1
dell-base-if-cmn/set-interface/input/operation = 1
base-if-phy/hardware-port/front-panel-port = 15
base-if-phy/if/interfaces/interface/hw-profile = 3
cps/object-group/return-code = 0
base-if-phy/hardware-port/subport-id = 4
dell-if/if/interfaces/interface/speed = 4
base-if-phy/if/interfaces/interface/port-id = 116
base-if-phy/if/interfaces/interface/phy-media = 43
base-if-phy/if/interfaces/interface/supported-autoneg = 4
dell-if/if/interfaces/interface/negotiation = 1
root@OPX:~#
root@OPX:~#
root@OPX:~# opx-ethtool e101-015-1
Settings for e101-015-1:
    Channel ID:   1
    Transceiver Status: Disable
    Media Type: QSFP 40GBASE CR4
    Part Number: L56QF031-SD-R
    Serial Number: 171361393
    Qualified: Yes
    Administrative State: DOWN
    Operational State: DOWN
    Supported Speed (in Mbps):  [10000]
    Auto Negotiation : on
    Configured Speed   : 10000
    Operating Speed   : 0
    Duplex   : full
root@OPX:~# ip link set dev e101-015-1 up
root@OPX:~# opx-ethtool e101-015-1
Settings for e101-015-1:
    Channel ID:   1
    Transceiver Status: Disable
    Media Type: QSFP 40GBASE CR4
    Part Number: L56QF031-SD-R
    Serial Number: 171361393
    Qualified: Yes
    Administrative State: UP
    Operational State: DOWN
    Supported Speed (in Mbps):  [10000]
    Auto Negotiation : on
    Configured Speed   : 10000
    Operating Speed   : 0
    Duplex   : full
root@OPX:~# opx-ethtool e101-015-1
Settings for e101-015-1:
    Channel ID:   1
    Transceiver Status: Disable
    Media Type: QSFP 40GBASE CR4
    Part Number: L56QF031-SD-R
    Serial Number: 171361393
    Qualified: Yes
    Administrative State: UP
    Operational State: UP
    Supported Speed (in Mbps):  [10000]
    Auto Negotiation : on
    Configured Speed   : 10000
    Operating Speed   : False
    Duplex   : full
root@OPX:~#
```
There are two issues with the above approach:
(1) It is very inconvenient. After doing opx-config-hybrid-group, having to do a `nas_intf_set.py` call for every breakout interface is a bad experience. Either the `opx-config-hybrid-group` command should just take care of interface deletion/creation as appropriate, or `opx-config-fanout` should take care of it as an additional step

(2) There needs to be a way to persist the fanout configuration on the Z9264F so that hybrid ports are configured and fanned out on boot.

**Comment by jeff-yin** [comment]
This will be addressed in the next release of OPX

**Comment by jeff-yin** [comment]
Resolved by #64 and https://github.com/open-switch/opx-nas-interface/pull/61/files
